### PR TITLE
Add name validation to food

### PR DESCRIPTION
**Pull request by TheKaterTot**
This fixes the unhandled promise rejection warnings, and adds name validations for foods. Records are not made in the database, and an error is raised. Fully tested and refactored to the best of my ability at the moment.

### CODE REVIEW

File: lib/controllers/foods-controller.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ function index (req, res) {
 };
 
 function create (req, res) {
-  Food.create(req)
+  Food.create(req.body.food.name, req.body.food.calories)
```
Comment (cmacaulay) [inline lib/controllers/foods-controller.js:11]: 🎉 It's super helpful to know about separation of concerns! @TheKaterTot 
Do you think we want to save these as constants and then pass them through, any idea on convention on whether or not to do that? #JSNoob

File: lib/models/food.js [hunk 2]
```diff
@@ -6,17 +6,24 @@ function all(){
   return database.raw('SELECT * FROM foods')
 };
 
-function create(req) {
+function create(name, calories) {
   return new Promise( (resolve, reject) => {
-    const food = req.body.food
-    validation(food, (err) => {
+
+    calorieValidation(calories, (err) => {
```
Comment (cmacaulay) [inline lib/models/food.js:12]: 👏 👏

File: lib/models/food.js [hunk 3]
```diff
@@ -6,17 +6,24 @@ function all(){
...
+      nameValidation(name, (err) => {
+        if (err) {
+          return reject(err);
+        }
+
+        database.raw('INSERT INTO foods (name, calories, created_at) VALUES (?, ?, ?) RETURNING id, name, calories',
+          [name, calories, new Date])
+            .then(resolve)
+            .catch(reject)
+      })
```
Comment (cmacaulay) [inline lib/models/food.js:26]: 👏 promises!

File: lib/models/food.js [hunk 4]
```diff
@@ -25,13 +32,20 @@ function destroy(req) {
...
   callback();
 }
 
+function nameValidation(name, callback) {
+  if (!name) {
+    return callback(new Error('Food must have name.'));
+  }
+  callback();
+}
+
```
Comment (cmacaulay) [inline lib/models/food.js:48]: 🎉 🎉 I think that having these in the backend is going to make implementing the front end go smoothly.

File: test/food-test.js [hunk 5]
```diff
@@ -6,11 +6,20 @@ const Food = require('../lib/models/food');
...
-    done()
+    done();
+  })
+
+  it('is invalid without name', (done) => {
+    Food.create(null, 40)
+      .catch( (err) => {
+        assert.equal(err.message, 'Food must have name.')
+      })
+
```
Comment (cmacaulay) [inline test/food-test.js:22]: thanks for increasing the test coverage on this!

**Review by cmacaulay — APPROVED** [review]
🎉 🎉 🎉

File: lib/controllers/foods-controller.js (hunk 1 above)
Comment (TheKaterTot) [reply lib/controllers/foods-controller.js:11]: I think we might want to save them in an object. Like:

let food = {name: req.body.food.name,
                   calories: req.body.food.calories, etc.}

**Pull request merged**